site/src/modules/resources/AgentLogLine.tsx

```diff
@@ -0,0 +1,60 @@
+import type { Interpolation, Theme } from "@emotion/react";
+import AnsiToHTML from "ansi-to-html";
+import { type FC, type ReactNode, useMemo } from "react";
+import {
+  type Line,
+  LogLine,
+  LogLinePrefix,
+  LogLineSpace,
+} from "components/Logs/LogLine";
+
+const convert = new AnsiToHTML();
+
+interface AgentLogLineProps {
+  line: Line;
+  number: number;
+  style: React.CSSProperties;
+  sourceIcon: ReactNode;
+  maxNumber: number;
+}
+
+export const AgentLogLine: FC<AgentLogLineProps> = ({
+  line,
+  number,
+  maxNumber,
+  sourceIcon,
+  style,
+}) => {
+  const output = useMemo(() => {
+    return convert.toHtml(line.output.split(/\r/g).pop() as string);
+  }, [line.output]);
+
+  return (
+    <LogLine css={{ paddingLeft: 16 }} level={line.level} style={style}>
+      {sourceIcon}
+      <LogLinePrefix
+        css={styles.number}
+        style={{
+          minWidth: `${maxNumber ? maxNumber.toString().length - 1 : 0}em`,
+        }}
+      >
+        {number}
+      </LogLinePrefix>
+      <LogLineSpace />
+      <span
+        dangerouslySetInnerHTML={{
+          __html: output,
+        }}
```

Comment:
why is this necessary? `dangerouslySetInnerHTML` should always be accompanied by an explaination

Reply:
Can we just add a comment above this that says "`output` contains HTML to represent ANSI-code formatting" or something then?

Reply:
Sure thing!

Reply:
Done!